### PR TITLE
Markers/tsnr

### LINKED ISSUE #163: [MARKER]: tSNR for BOLD
### Which marker do you want to include?

It will be a good marker for quality control.

### Is there any publication or available code?

voxelwise `mean/std` over timepoints

### Do you have a sample code that implements this outside of junifer?

_No response_

### Anything else to say?

_No response_

### PR DESCRIPTION
**Pull request by LeSasse**
* [x] fix #163 
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry to the [latest changes](../docs/changes/latest.inc)

Add a sub-package to compute voxelwise, parcelwise, and spherewise temporal signal-to-noise ratio.

### CODE REVIEW

File: junifer/markers/temporal_snr/temporal_snr_base.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# Authors: Leonard Sasse <l.sasse@fz-juelich.de>
+# License: AGPL
+
+
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional, Union
+
+from ...utils import raise_error
+from ..base import BaseMarker
+from ..utils import _voxelwise_tsnr
```
Comment (fraimondo) [inline junifer/markers/temporal_snr/temporal_snr_base.py:12]: why is this in utils? Isn't it only from this method?

File: junifer/markers/temporal_snr/temporal_snr_parcels.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        None).
+
+    """
+
+    def __init__(
+        self,
+        parcellation: Union[str, List[str]],
+        agg_method: str = "mean",
+        agg_method_params: Optional[Dict] = None,
+        masks: Optional[str] = None,
```
Comment (fraimondo) [inline junifer/markers/temporal_snr/temporal_snr_parcels.py:43]: outdated definition of masks.

File: junifer/markers/utils.py [hunk 3]
```diff
@@ -156,3 +157,34 @@ def _correlate_dataframes(
         .corr(method=method)  # type: ignore
         .loc["df2", "df1"]
     )
+
+
+def _voxelwise_tsnr(input: Dict) -> Dict:
```
Comment (fraimondo) [inline junifer/markers/utils.py:162]: no need to have this in utils, it can be in the base marker, making the TNSR marker self-contained.

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: junifer/markers/temporal_snr/temporal_snr_base.py (hunk 1 above)
Comment (synchon) [reply junifer/markers/temporal_snr/temporal_snr_base.py:12]: It seemed like the place to keep when the PR was started, now of course, it makes sense it in it's own submodule.

File: junifer/markers/temporal_snr/temporal_snr_parcels.py (hunk 2 above)
Comment (synchon) [reply junifer/markers/temporal_snr/temporal_snr_parcels.py:43]: Oops, missed it.

File: junifer/markers/temporal_snr/temporal_snr_base.py (hunk 1 above)
Comment (synchon) [reply junifer/markers/temporal_snr/temporal_snr_base.py]: Updated.

File: junifer/markers/utils.py (hunk 3 above)
Comment (synchon) [reply junifer/markers/utils.py]: Updated.

**Review by fraimondo — CHANGES_REQUESTED** [review]
Missing update in the docs: https://juaml.github.io/junifer/pr-preview/pr-171/builtin.html

**Comment by synchon** [comment]
> Missing update in the docs: https://juaml.github.io/junifer/pr-preview/pr-171/builtin.html

Done now.

**Review by fraimondo — APPROVED** [review]

**Comment by fraimondo** [comment]
LGTM!

**Pull request merged**